gamemodes/terrortown/gamemode/server/sv_network_sync.lua

```diff
@@ -0,0 +1,677 @@
+--
+-- @param any|table path The path this meta data is associated with (already includes the needed keywords)
+-- @param table|nil metadata The metadata table
+function TTT2NET:SetMetaDataOnPlayer(path, metadata, ply)
+	-- Convert path with single key to table
+	if not istable(path) then
+		path = { path }
+	end
+
+	-- Add the prefix for the correct table with the specific player
+	table.insert(path, 1, "players")
+	table.insert(path, 2, ply:EntIndex())
+
+	self:SetMetaData(path, metadata)
+end
+
+---
+-- This is used to set a value in the data table for a specific path, it can take an additional meta data object,
+-- which represents the type of the data to be set. This will automatically create the metadata entry if it does not
+-- exist and is provided as a parameter. This can only be left empty, when the path already has a valid metadata entry.
+-- This will automatically synchronize the new data to the clients.
+-- When the additional client parameter is set, the data will be saved as an override for that specific client/entity.
+--
+-- @param any|table path The path to set the data for
+-- @param table|nil meta The metadata for the path
+-- @param any value The value to save
+-- @param Entity|nil client The client/entity to set this value for (overrides the default value)
+function TTT2NET:Set(path, meta, value, client)
+	local clientId = client and client:EntIndex() or nil
+
+	-- Convert path with single key to table
+	if not istable(path) then
+		path = { path }
+	end
+
+	local metadata = meta or table.GetWithPath(data_store_metadata, path)
+	assert(metadata, "[TTT2NET] Set() called but no metadata entry or metadata parameter found!")
+
+	-- Set the meta data, this will only send / update if the meta data has changed
+	self:SetMetaData(path, metadata)
+
+	-- Add the identifier for the default value for all clients or the clientId if specified
+	path[#path + 1] = clientId or "default"
+
+	-- Skip if the value is already present or nil
+	if table.GetWithPath(data_store, path) == value then return end
+
+	-- Save the value
+	table.SetWithPath(data_store, path, value)
+
+	-- Remove last key eg. "default" or the player id, as this does not belong to the base path
+	path[#path] = nil
+
+	-- Sync the new value
+	self:SendDataUpdate(path, client)
+end
+
+---
+-- This will be used to clear out all client specific overrides for a path, so that only the
+-- default value is set.
+-- This will also automatically sync the new value to the clients that previously had an override value.
+--
+-- @param table path The path to clear out all overrides on.
+function TTT2NET:RemoveOverrides(path)
+	-- Convert single key to table
+	if not istable(path) then
+		path = { path }
+	end
+
+	local currentDataTable = data_store
+
+	-- Traverse the path in the data_store table.
+	-- This will be done until the second last table is reached, so we still have a reference to the parent table.
+	for i = 1, (#path - 1) do
+		if currentDataTable == nil then return end
```

Comment:
CS: missing empty line

Reply:
I would move the `if currentDataTable == nil then return end` into the end of the loop and the next currentDataTable check from below of this loop in front of this loop right below the declaration (to avoid entering useless loop). Just a small thing